core/src/UserCred.cs

```diff
@@ -0,0 +1,100 @@
+//------------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using System.Runtime.InteropServices;
+using System.Security;
+
+namespace Microsoft.Identity.Core
+{
+    internal enum UserAuthType
+    {
+        IntegratedAuth,
+        UsernamePassword
+    }
+
+    /// <summary>
+    /// Credential used for integrated authentication on domain-joined machines.
+    /// The implementation is  here, and the ADAL UserCredential class will subclass this.
+    /// </summary>
+    public class UserCred
```

Comment:
This is now a very thin base class. I wonder if you could use an interface here instead ?

Reply:
The plan is to have the identical implementation in Core, so that ADAL and MSAL would have a (possibly thin) sub-class. This could be changed if later we would find out MSAL does not need to use this class at all. I would suggest we postpone this discussion until later Username-and-Password flow implementation in MSAL.